spec.md

```diff
@@ -1090,6 +1084,252 @@ message ControllerServiceCapability {
 
 If the plugin is unable to complete the ControllerGetCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+#### `CreateSnapshot`
+
+A Controller Plugin MUST implement this RPC call if it has `CREATE_DELETE_SNAPSHOT` controller capability.
+This RPC will be called by the CO to create a new snapshot from a source volume on behalf of a user.
+
+This operation MUST be idempotent.
+If a snapshot corresponding to the specified snapshot `name` already exists and is compatible with the specified `source_volume_id` and `parameters` in the `CreateSnapshotRequest`, the Plugin MUST reply `0 OK` with the corresponding `CreateSnapshotResponse`.
+
+Snapshot can be used as the source to provision a new volume.
```

Comment:
@xing-yang while we promote the snap to a new volume do we have a mechanism to say the new volume should be `READONLY` or ` READWRITE` ? or an implicit mention in snapshot response itself the new volumes will/can only be `READONLY` or `READWRITE` from this snap? may be I missed it.

Reply:
@humblec In CreateVolumeRequest, there is a VolumeCapability where you can define AccessMode.  Create volume from a snapshot and create a blank volume should work the same way there.

Reply:
@xing-yang cool, I missed that field. thanks!